web-api/src/mind.rs

```diff
@@ -231,6 +299,122 @@ where
         .map_err(Into::into)
 }
 
+// function that assigns a score to a vector of documents based on the user's interests
+// score is equal to 2 if the document is interesting to the user, 0 otherwise
+fn score_documents(documents: &[&Document], user_interests: &[String]) -> Array1<f32> {
+    documents
+        .iter()
+        .map(|document| {
+            if document.is_interesting(user_interests) {
+                2.0
+            } else {
+                0.0
+            }
+        })
+        .collect_vec()
+        .into()
+}
+
+fn write_array<T, S, D>(writer: impl io::Write, array: &ndarray::ArrayBase<S, D>) -> io::Result<()>
+where
+    T: Clone + npyz::AutoSerialize,
+    S: ndarray::Data<Elem = T>,
+    D: ndarray::Dimension,
+{
+    let shape = array.shape().iter().map(|&x| x as u64).collect_vec();
+    let c_order_items = array.iter();
+
+    let mut writer = npyz::WriteOptions::new()
+        .default_dtype()
+        .shape(&shape)
+        .writer(writer)
+        .begin_nd()?;
+    writer.extend(c_order_items)?;
+    writer.finish()
+}
+#[doc = "
+
+# Panics
+
+The function panics if the provided filenames are not correct.
+"]
```

Comment:
it's technically correct, but the more readable format is to use `///` instead of `#[doc = ""]`
```suggestion
/// # Panics
///
/// The function panics if the provided filenames are not correct.
```